tests/query_test.py

```diff
@@ -66,19 +66,45 @@ def test_quote(self):
     quoted = query.let({"x": 1}, query.var("x"))
     assert self._q(query.quote(quoted)) == quoted
 
+  def test_lambda_query(self):
+    assert query.lambda_query(lambda a: query.add(a, a)) == {
+      "lambda": "auto0", "expr": {"add": ({"var": "auto0"}, {"var": "auto0"})}
+    }
+
+    expected = query.lambda_query(
+      lambda a: query.lambda_query(
+        lambda b: query.lambda_query(
+          lambda c: [a, b, c])))
+    assert expected == {
+      "lambda": "auto0",
+      "expr": {
+        "lambda": "auto1",
+        "expr": {
+          "lambda": "auto2",
+          "expr": [{"var": "auto0"}, {"var": "auto1"}, {"var": "auto2"}]
+        }
+      }
+    }
+
+    # Error in lambda should not affect future queries.
+    def fail_in_lambda():
```

Comment:
I would like to see this function removed and leverage `assertRaises`, either by passing in `lambda_query` and it's lambda as an arg, or using it as a context manager.